File: practice/algorithms/implementation/viral_advertising/ViralAdvertising.js
'use strict';

const processData = input => {
    const n = parseInt(input)

    let totalLiked = 0;
    let dayStartPeople = 5;
    for(let i = 0; i < n; i++) {
        const dayLiked = parseInt(dayStartPeople / 2);
        const received = dayLiked * 3;
        totalLiked += dayLiked;
        dayStartPeople = received;
    }
    console.log(totalLiked);
};

process.stdin.resume();
process.stdin.setEncoding("ascii");

let _input = "";
process.stdin.on("data", input => _input += input);
process.stdin.on("end", () => processData(_input));

File: practice/algorithms/implementation/viral_advertising/viral_advertising.rb
n = gets.to_i
    
total_liked = 0
day_start_people = 5
for i in 0...n do
  day_liked = day_start_people / 2
  received = day_liked * 3
  total_liked = total_liked + day_liked
  day_start_people = received
end

puts total_liked